go/tasks/plugins/k8s/pod/plugin.go

```diff
@@ -184,8 +191,17 @@ func (plugin) GetTaskPhaseWithLogs(ctx context.Context, pluginContext k8s.Plugin
 		return pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion, &info), nil
 	}
 
+	// When the copilot is running, we should wait until the data is uploaded by the copilot.
+	copilotContainerName, exists := r.GetAnnotations()[flytek8s.FlyteCopilotName]
+	if exists {
+		copilotContainerPhase := flytek8s.DetermineContainerPhase(copilotContainerName, pod.Status.ContainerStatuses, &info)
+		if copilotContainerPhase.Phase() == pluginsCore.PhaseRunning && len(info.Logs) > 0 {
+			return pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion+1, copilotContainerPhase.Info()), nil
+		}
+	}
+
```

Comment:
So right now this is done in `ContainerTasks` (not using map task) by just not setting a `primaryContainerName` on the `Pod`, then this code waits for the entire `Pod` to complete. It seems like this is what we should do for subtasks as well.

I think the issue is that [here](https://github.com/flyteorg/flyteplugins/blob/0a681cd1a9e2c6586e6408532767539be65737f4/go/tasks/plugins/array/k8s/subtask.go#L325-L344) we always add a `PrimaryContainerName` to the pod annotation. I think it makes sense to update the code in the subtask.go so that this annotation is only added if necessary, then we shouldn't need to add to `flytek8s.FlyteCopilotName` annotation above either.

Reply:
The issue is the propeller will only pass array index to primary container, so we have to set raw-container to primary. However, if we set it to primary, propeller won't wait for the uploader complete, so I added `flytek8s.FlyteCopilotName` to annotation, and wait for copilot first if we find the uploader container in the pod here.